### ISSUE TITLE
Expo + Google OAuth, documentation incorrect / not working

### ISSUE THREAD

**Issue by williamlmao** [supabase/supabase#17999]
https://supabase.com/docs/guides/auth/social-login/auth-google?platform=react-native#using-native-sign-in-with-google-in-expo

I followed the docs above to a tee, and the google sign in IS working. I can console log the `userInfo` and see all of my login info from google, including my `userInfo.idToken`. 

However, I am getting an error `[AuthApiError: Passed nonce and nonce in id_token should either both exist or not.]` from these lines: 

```
  const { data, error } = await supabase.auth.signInWithIdToken({
              provider: 'google',
              token: userInfo.idToken,
              
            })
            console.log(error, data)
```
            
The main docs mention generating a nonce, but the expo specific docs do not. I tried to follow a similar flow as the expo + apple auth tutorial, but we can't actually pass a nonce into the google sign in. 

Additionally, see this thread https://github.com/react-native-google-signin/google-signin/issues/1176

Does anyone have any working examples of Google OAuth in an Expo app? Does it require a nonce or not because the documentation is showing it one way but the error is asking for another.

**Comment by Quilljou** [comment]
same here

**Comment by dthomason** [comment]
apparently they are working on a fix, https://github.com/supabase/gotrue/issues/1205#issuecomment-1730933104
haven't seen anything released for supabase-js quite yet.  Hopefully soon.

**Comment by polooner** [comment]
merged in #18301 btw if you want to close
